ToolChain/Debugger/Debugger.py

```diff
@@ -592,6 +625,18 @@ def addBreakPoint(self, address=None):
         self.breakPoints.append(address)
         self.breakPoints.sort()
 
+        self.writeBreakPointFile()
+
+    def writeBreakPointFile(self):
+        """
+        This function writes the breakpoints stored in self.breakpoints to a file
+        :return:
+        """
+        file = open((self.inputFile.split(".")[0]+".bp"), "w")
```

Comment:
Python open can generate exceptions. 
https://docs.python.org/3/library/functions.html#open
I'm aware that the other opens in this file don't take this into account but we should fix it (this one as well as the other).